core/frontend/src/types/helper.ts

```diff
@@ -1,7 +1,18 @@
+export interface ServiceMetadata {
+    name: string
+    description: string
+    icon: string
+    company: string
+    version: string
+    webpage: string
+    api: string
+}
+
 export interface Service {
     valid: boolean
     title: string
     documentation_url: string
     versions: Array<string>
     port: number
+    metadata?: ServiceMetadata
 }
```

Comment:
Regarding this, how should one creating a service behave in respect to `name` vs `title`, `valid`, `webpage` vs `api` vs `documentation_url`, for example?

Reply:
Service is a structure that we use to help us easily identify services with the helper scraping logic. the metadata is the real information provided by the running service. 
We should rely on the metadata to provide all the necessary information for the frontend. Service was only a quick way to make it work without having this concept finished.
Remember that this is still not a supported thing and we can rework everything around this concept before have official support. The PR only allows the metadata access for people and services that are willing to have it working right now and accept the API break in the future.